nexus/db-queries/src/db/pool_connection.rs

```diff
@@ -4,46 +4,91 @@
 
 //! Customization that happens on each connection as they're acquired.
 
+use anyhow::anyhow;
+use async_bb8_diesel::AsyncR2D2Connection;
 use async_bb8_diesel::AsyncSimpleConnection;
-use async_bb8_diesel::Connection;
-use async_bb8_diesel::ConnectionError;
 use async_trait::async_trait;
-use bb8::CustomizeConnection;
+use diesel::Connection;
 use diesel::PgConnection;
 use diesel_dtrace::DTraceConnection;
+use qorb::backend::{self, Backend, Error};
 
 pub type DbConnection = DTraceConnection<PgConnection>;
 
 pub const DISALLOW_FULL_TABLE_SCAN_SQL: &str =
     "set disallow_full_table_scans = on; set large_full_scan_rows = 0;";
 
-/// A customizer for all new connections made to CockroachDB, from Diesel.
-#[derive(Debug)]
-pub(crate) struct ConnectionCustomizer {}
+/// A [backend::Connector] which provides access to [PgConnection].
+pub struct DieselPgConnector {
```

Comment:
does this need to be public? or is it internal to `db-queries`?

Reply:
I can make it crate-local! Done